Topic3/src/org/maugiam/tdd/main/RecentFileList.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * 
+ */
+package org.maugiam.tdd.main;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * @author Mauro J Giamberardino
+ *
+ */
+public class RecentFileList {
+
+	protected Integer limit;
+	protected LinkedList<String> fileList;
+
+	public RecentFileList(Integer limit) {
+		this.fileList = new LinkedList<String>();
+		this.limit = limit;
+	}
+
+	public List<String> getRecentList() {
+		return fileList;
+	}
+
+	public void addOpenedFile(String filePath) {
+		int index = fileList.indexOf(filePath);
+		if (index < 0) {
+			if (fileList.size() >= limit)

```

Comment:
Warning: Use {} for if statements. Here is the perfect example why to use it. if you may have forgotten to properly ndent the fileList.push line, someone may think that line is still part of the if statement when it is not. As the owner of the code you probably won't overlook that, but consider that class may be maintained by others in the future. That is why the {} practice is encouraged.